### PR TITLE
[thermal fix] Fix issues for thermal control test cases

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
### Description of PR

Summary:
Fixes # (issue)
1. test_thermal_control_psu_absence is always skipped
2. PSU speed should be no less than 100

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### How did you verify/test it?

Run regression

### CODE REVIEW

File: tests/platform/test_platform_info.py [hunk 1]
```diff
@@ -441,12 +441,12 @@ def test_thermal_control_psu_absence(testbed_devices, psu_controller, mocker_fac
...
             pytest.skip("No FanStatusMocker for %s, skip rest of the testing in this case" % dut.facts['asic_type'])
 
-        logging.info('Mock FAN status data...')
-        fan_mocker.mock_data()  # make data random
         restart_thermal_control_daemon(dut)
         logging.info('Wait and check all FAN speed turn to 60%...')
-        wait_until(THERMAL_CONTROL_TEST_WAIT_TIME, THERMAL_CONTROL_TEST_CHECK_INTERVAL, fan_mocker.check_all_fan_speed,
-                   60)
+        wait_result = wait_until(THERMAL_CONTROL_TEST_WAIT_TIME, THERMAL_CONTROL_TEST_CHECK_INTERVAL, fan_mocker.check_all_fan_speed,
+                                 60)
```
Comment (wangxin) [inline tests/platform/test_platform_info.py]: Please correct the format issue.

File: tests/platform/test_platform_info.py [hunk 2]
```diff
@@ -466,8 +466,8 @@ def test_thermal_control_psu_absence(testbed_devices, psu_controller, mocker_fac
             pytest.skip("Some PSU are still down, skip rest of the testing in this case")
 
         logging.info('Wait and check all FAN speed turn to 65%...')
-        wait_until(THERMAL_CONTROL_TEST_WAIT_TIME, THERMAL_CONTROL_TEST_CHECK_INTERVAL, fan_mocker.check_all_fan_speed,
-                   65)
+        assert wait_until(THERMAL_CONTROL_TEST_WAIT_TIME, THERMAL_CONTROL_TEST_CHECK_INTERVAL, fan_mocker.check_all_fan_speed,
+                          65), 'FAN speed not change to 65% according to policy'
```
Comment (wangxin) [inline tests/platform/test_platform_info.py]: Same format issue here.

File: tests/platform/test_platform_info.py [hunk 3]
```diff
@@ -490,9 +490,11 @@ def turn_off_psu_and_check_thermal_control(dut, psu_ctrl, psu, mocker):
 
     assert psu_under_test is not None, "No PSU is turned off"
     logging.info('Wait and check all FAN speed turn to 100%...')
-    wait_until(THERMAL_CONTROL_TEST_WAIT_TIME, THERMAL_CONTROL_TEST_CHECK_INTERVAL, mocker.check_all_fan_speed, 100)
+    assert wait_until(THERMAL_CONTROL_TEST_WAIT_TIME, THERMAL_CONTROL_TEST_CHECK_INTERVAL, mocker.check_all_fan_speed, 
+                      100), 'FAN speed not turn to 100% after PSU off'
```
Comment (wangxin) [inline tests/platform/test_platform_info.py]: Same format issue here.

**Comment by wangxin** [comment]
Content of below two functions are same:
* test_thermal_control_load_invalid_format_json
* test_thermal_control_load_invalid_value_json
They called check_thermal_control_load_invalid_file with same arguments:
```
    check_thermal_control_load_invalid_file(testbed_devices, THERMAL_POLICY_INVALID_VALUE_FILE)
```
For the `test_thermal_control_load_invalid_format_json`, argument THERMAL_POLICY_INVALID_FORMAT_FILE should be used.

**Comment by wangxin** [comment]
Line 368-369 in test_platform_info.py:
```
    cli_thermal_status = dut.command(CMD_PLATFORM_TEMPER)
    assert cli_thermal_status["rc"] == 0, "Run command '%s' failed" % CMD_PLATFORM_TEMPER
```
If 'rc' of dut.command is not 0, dut.command itself will raise exception unless it has keyword argument 'module_ignore_errors=True'. So, the `assert` line would not be hit if dut.command failed.

Please refer to the implementation of tests/common/devices.py::AnsibleHostBase._run for more details.

**Comment by Junchao-Mellanox** [comment]
Hi @wangxin, if the command execution failed, raising exception and stopping the test case is also good for me, i don't want to ignore the error.

**Comment by wangxin** [comment]
I understand. The `assert` line would be useless. When rc==0, the assert line will do nothing. When rc!=0, the assert line will not be hit. Suggest to remove the `assert` line.

**Comment by Junchao-Mellanox** [comment]
Community PR: https://github.com/Azure/sonic-mgmt/pull/1524

**Pull request closed without merging**